gobblin-cluster/src/main/java/org/apache/gobblin/cluster/FsScheduledJobConfigurationManager.java

```diff
@@ -53,27 +51,26 @@ public FsScheduledJobConfigurationManager(EventBus eventBus, Config config, Muta
 
   @Override
   protected void fetchJobSpecs() throws ExecutionException, InterruptedException {
-    List<Pair<SpecExecutor.Verb, Spec>> jobSpecs =
-        (List<Pair<SpecExecutor.Verb, Spec>>) this._specConsumer.changedSpecs().get();
+    List<Pair<SpecExecutor.Verb, JobSpec>> jobSpecs =
```

Comment:
Should make sure that the job configuration manager receives the specs in modification time order.

Reply:
Done.